src/java/org/apache/cassandra/cql3/statements/schema/CreateTableStatement.java

```diff
@@ -118,6 +122,36 @@ public Keyspaces apply(Keyspaces schema)
         return schema.withAddedOrUpdated(keyspace.withSwapped(keyspace.tables.with(table)));
     }
 
+    @Override
+    public void validate(ClientState state)
+    {
+        super.validate(state);
+
+        // Some tools use CreateTableStatement, and the guardrails below both don't make too much sense for tools and
+        // require the server to be initialized, so skipping them if it isn't.
+        if (Guardrails.enabled())
```

Comment:
I agree they don't really make sense for tools, but this check holds true even for actual server case; if not enabled do less work.

Each check below will check for enabled (volatile load) as well as feature/warn/fail enabled (volatile load), so this check is just an optimization for the case where guardrails are disabled.

Reply:
Yes, it's a trivial optimization to avoid doing multiple calls to `Guardrail#enabled` and calculating the parameters.

Reply:
Can we update the comment (or remove)?  If I look at it 1y from now I would think there is some dragon lurking in the dark that this protects against, rather than this is an optimization when the feature is turned off.

Reply:
+1, I have updated the comment